scripts/generate_psa_key_attributes.py

```diff
@@ -0,0 +1,238 @@
+        allow_abbrev=False,
+    )
+
+    parser.add_argument(
+        "--usage",
+        help="Key usage",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaUsage],
+    )
+
+    parser.add_argument(
+        "--id",
+        help="Key identifier",
+        type=lambda number_string: int(number_string, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--type",
+        help="Key type",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaKeyType],
+    )
+
+    parser.add_argument(
+        "--size",
+        help="Key size in bits",
+        type=lambda number_string: int(number_string, 0),
+        required=True,
+    )
+
+    parser.add_argument(
+        "--algorithm",
+        help="Ket algorithm",
+        type=str,
+        required=False,
+        default="NONE",
+        choices=[x.name for x in PsaAlgorithm],
+    )
+
+    parser.add_argument(
+        "--location",
+        help="Storage location",
+        type=str,
+        required=True,
+        choices=[x.name for x in PsaKeyLifetime],
+    )
+
+    parser.add_argument(
+        "--key",
+        help="Key value",
+        type=str,
+        required=True,
+    )
+
+    parser.add_argument(
+        "--file",
+        help="JSON file to create or modify",
+        type=str,
+        required=False,
+    )
+
+    parser.add_argument(
+        "--cracen_usage",
+        help="CRACEN KMU Slot usage scheme",
+        type=str,
+        required=False,
+        default="NONE",
+        choices=[x.name for x in PsaCracenUsageSceme],
+    )
+
+    args = parser.parse_args()
+
+    metadata = binascii.hexlify(PlatformKeyAttributes(key_type=PsaKeyType[args.type],
+                              identifier=args.id,
+                              location=PsaKeyLifetime[args.location],
+                              usage=PsaUsage[args.usage],
+                              algorithm=PsaAlgorithm[args.algorithm],
+                              size=args.size,
+                              cracen_usage=PsaCracenUsageSceme[args.cracen_usage])
+                                .pack()).decode('utf-8').upper()
+
+    if args.file and Path(args.file).is_file():
+        with open(args.file, encoding="utf-8") as file:
+            json_data= json.load(file)
+    else:
+        json_data= json.loads('{ "version": 0, "keyslots": [ ]}')
+
+
+    if args.key != "TRNG":
+        key = args.key.strip("0x")
+        if not is_valid_hexa_code(key):
+            print("Invalid KEY value")
+            return
+        value = f'0x{key}'
+    else:
+        key = args.key
+        value  = f'{key}:{int(math.ceil(args.size / 8))}'
+
+    json_data["keyslots"].append({"metadata": f'0x{metadata}00000000', "value": f'{value}'})
+    output = json.dumps(json_data, indent=4)
+
+    if args.file:
+        with open(args.file, "w", encoding="utf-8") as file:
+            file.write(output)
+    else:
+        print(output)
```

Comment:
Should it be conditional if we generate the json format or instead only the binary blob? That way the script could be re-used for other purposes where the PSA attribute binary blob is needed.

Reply:
Ok